packages/data-mate/src/transforms/field-transform.ts

```diff
@@ -271,91 +455,421 @@ export function toISDN(input: any) {
+    if (ts.isNil(input) && args?.booleanLike !== true) return null;
+
+    if (isArray(input)) {
+        return input
+            .filter(ts.isNotNil)
+            .map((data: any) => convertToNumber(data, args));
+    }
+
+    return convertToNumber(input, args);
+}
+
+/**
+ * decodes a base64 hashed value
+ * if given an array it will convert everything in the array excluding null/undefined values
+ *
+ * @example
+ * const str = 'hello world';
+ * const encoded = encodeBase64(str);
+ *
+ * const results = fieldTransform.decodeBase64(encoded)
+ * results === str
+ *
+ * @export
+ * @param {*} input
+ * @returns { string | null } returns null if input is null/undefined
+ */
+
 export function decodeBase64(input: any) {
     if (ts.isNil(input)) return null;
+
+    if (isArray(input)) {
+        return input
+            .filter(ts.isNotNil)
+            .map((data: any) => Buffer.from(data, 'base64').toString('utf8'));
+    }
+
     return Buffer.from(input, 'base64').toString('utf8');
 }
 
+/**
+ * encodes a value into base64
+ * if given an array it will convert everything in the array excluding null/undefined values
+ *
+ * @example
+ * const str = 'hello world';
+ *
+ * const encodedValue = fieldTransform.decodeBase64(str);
+ *
+ * @export
+ * @param {*} input
+ * @returns { string | null } returns null if input is null/undefined
+ */
+
 export function encodeBase64(input: any) {
     if (ts.isNil(input)) return null;
+
+    if (isArray(input)) {
+        return input
+            .filter(ts.isNotNil)
+            .map((data: any) => Buffer.from(data).toString('base64'));
+    }
+
     return Buffer.from(input).toString('base64');
 }
 
-export function decodeUrl(input: string) {
+/**
+ * decodes a url encoded value
+ * if given an array it will convert everything in the array excluding null/undefined values
+ *
+ * @example
+ * const source = 'HELLO AND GOODBYE';
+ * const encoded = 'HELLO%20AND%20GOODBYE';
+ *
+ * expect(fieldTransform.decodeUrl(encoded)).toEqual(source);
+ *
+ * @export
+ * @param {StringInput} input
+ * @returns { string | null } returns null if input is null/undefined
+ */
+
+export function decodeUrl(input: StringInput) {
     if (ts.isNil(input)) return null;
+
+    if (isArray(input)) return input.filter(ts.isNotNil).map(decodeURIComponent);
+    if (!isString(input)) throw new Error(`Input must be a string, or an array of string, received ${ts.getTypeOf(input)}`);
+
     return decodeURIComponent(input);
 }
 
-export function encodeUrl(input: string) {
+/**
+ * url encodes a value
+ * if given an array it will convert everything in the array excluding null/undefined values
+ *
+ * @example
+ * const source = 'HELLO AND GOODBYE';
+ * const encoded = 'HELLO%20AND%20GOODBYE';
+ *
+ * expect(fieldTransform.encodeUrl(source)).toEqual(encoded);
+ *
+ * @export
+ * @param {StringInput} input
+ * @returns { string | null } returns null if input is null/undefined
+ */
+
+export function encodeUrl(input: StringInput) {
```

Comment:
We should be consistent with naming:

```suggestion
export function encodeURL(input: StringInput) {
```